### PR TITLE
Service discovery

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- fetch-queue won't work for `core` - it somehow breaks the e2e tests ([checked here](https://github.com/kyma-project/busola/pull/1320)), even with concurrency bumped from 3 to 8.

**Todo**

* Cache the data in the storage type used by the cluster (local/session/memory) - for now it's only kept in `localStorage`.

### CODE REVIEW

**Comment by Wawrzyn321** [comment]
We'll wait for https://github.com/kyma-project/busola/issues/700 and reopen this again. We left some todos, instructions and comments in the code, so work can be easily resumed.

File: docs/features.md [hunk 1]
```diff
@@ -2,7 +2,19 @@
 title: Feature flags
 ---
 
-The document lists and describes all the feature flags that are available in Kyma Dashboard and provides their configuration examples:
+The document explains the usage of feature flags in Busola, as well as lists and describes all the feature flags that are available in Kyma Dashboard and provides their configuration examples:
```
Comment (grego952) [inline docs/features.md:5]:
```suggestion
The document explains the usage of feature flags in Busola, lists and describes all the available feature flags, and provides their configuration examples:
```
Maybe we can shorten the intro like above?

File: docs/features.md [hunk 2]
```diff
@@ -2,7 +2,19 @@
...
+
+#### Features priority
+
+Features in Busola initialisation is based on `stage` property, which can take one of the values listed here:
```
Comment (grego952) [inline docs/features.md:9]:
```suggestion
Initialisation of the Busola features is based on the `stage` property, which can take one of the following values:
```

File: docs/features.md [hunk 3]
```diff
@@ -2,7 +2,19 @@
...
+
+- `PRIMARY` - the feature will be resolved while the app bootstraps. Features that should be immediately visible should be set as `PRIMARY` (e.g. main navigation structure).
```
Comment (grego952) [inline docs/features.md:11]:
```suggestion
- `PRIMARY` - the feature is resolved while the application bootstraps. Features that should be immediately visible must be set as `PRIMARY` (for example, main navigation structure).
```

File: docs/features.md [hunk 4]
```diff
@@ -2,7 +2,19 @@
...
+- `SECONDARY` - the feature will be resolved after the app is ready, it should be used for non-critical features (e.g. additional navigation nodes).
```
Comment (grego952) [inline docs/features.md:12]:
```suggestion
- `SECONDARY` - the feature is resolved after the application is ready, it must be used for non-critical features (for example, additional navigation nodes).
```

File: docs/features.md [hunk 5]
```diff
@@ -2,7 +2,19 @@
...
+- \<other value> - the feature will be loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.
```
Comment (grego952) [inline docs/features.md:13]:
```suggestion
- \<other value> - the feature is loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.
```

File: docs/features.md [hunk 6]
```diff
@@ -2,7 +2,19 @@
...
+
+Note that some features need to be run before the app starts the bootstrap process (e.g. SSO_LOGIN), so they are out of normal feature flow.
```
Comment (grego952) [inline docs/features.md:15]:
```suggestion
Note that some features must be run before the application starts the bootstrap process (for example, SSO_LOGIN), so they are out of the normal feature flow.
```

File: core/src/luigi-config/feature-checks.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import { fetchCache } from './cache/fetch-cache';
+import { updateFeature } from './feature-discovery';
+import { getAuthData } from './auth/auth-storage';
+import { extractGroupVersions } from './utils/extractGroupVersions';
+
+export function apiGroup(group, refreshIntervalMs = 5 * 60 * 1000) {
```
Comment (qbalukom) [inline core/src/luigi-config/feature-checks.js:6]: maybe go with named parameters instead? to keep it consistent with `service()`?

**Review by qbalukom — CHANGES_REQUESTED** [review]

File: core/src/luigi-config/cluster-management/cluster-management.js [hunk 8]
```diff
@@ -104,7 +107,7 @@ export async function saveClusterParams(params) {
 export async function saveCARequired() {
   const clusters = clusterStorage.load();
   const cluster = clusters[getActiveClusterName()];
-  if (cluster.config.requiresCA === undefined) {
+  if (cluster?.config.requiresCA === undefined) {
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply core/src/luigi-config/cluster-management/cluster-management.js:110]: I heard about it, but is it still possible in modern browsers? 👀

[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply core/src/luigi-config/cluster-management/cluster-management.js:110]: Woah, it is actually possible, but only in IIFE (https://stackoverflow.com/a/8783528/9462061).
It doesn't work in browser console though
image

I'll edit the code anyway.

**Review by qbalukom — APPROVED** [review]

File: docs/features.md [hunk 9]
```diff
@@ -2,7 +2,19 @@
...
-The document lists and describes all the feature flags that are available in Kyma Dashboard and provides their configuration examples:
+The document explains the usage of feature flags in Busola, lists and describes all the available feature flags, and provides their configuration examples:
+
+#### Features priority
+
+Initialisation of the Busola features is based on the `stage` property, which can take one of the following values:
+
+- `PRIMARY` - the feature is resolved while the application bootstraps. Features that should be immediately visible must be set as `PRIMARY` (for example, main navigation structure).
+- `SECONDARY` - the feature is resolved after the application is ready, it must be used for non-critical features (for example, additional navigation nodes).
+- \<other value> - the feature is loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.
```
Comment (grego952) [inline docs/features.md:13]:
```suggestion
- `{other_value} - the feature is loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.
```

**Review by grego952 — APPROVED** [review]

**Review by qbalukom — APPROVED** [review]

**Pull request merged**